packages/blade/src/components/Link/BaseLink/__tests__/BaseLink.native.test.tsx

```diff
@@ -0,0 +1,380 @@
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="information" contrast="high">
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render disabled information intent high contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="information" contrast="high" isDisabled={true}>
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render notice intent low contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="notice" contrast="low">
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render disabled notice intent low contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="notice" contrast="low" isDisabled={true}>
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render notice intent high contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="notice" contrast="high">
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render disabled notice intent high contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="notice" contrast="high" isDisabled={true}>
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render neutral intent low contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="neutral" contrast="low">
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render disabled neutral intent low contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="neutral" contrast="low" isDisabled={true}>
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render neutral intent high contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="neutral" contrast="high">
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render disabled neutral intent high contrast link', () => {
+    const linkText = 'Learn More';
+    const { toJSON } = renderWithTheme(
+      <BaseLink intent="neutral" contrast="high" isDisabled={true}>
+        {linkText}
+      </BaseLink>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
```

Comment:
Hahaha lets have a separate discussion around this. I know these tests seem a lot but I'm quite scared of future changes breaking existing visuals/functionalities especially since these components would be widely used across projects